### PR TITLE
Updsate the zipfile format to prevent choking on json parsing

### PR DESCRIPTION
**Pull request by kbirk**

### CODE REVIEW

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/AssetExportDeserializer.java [hunk 1]
```diff
@@ -14,21 +14,21 @@
...
 
 		final String assetTypeStr = node.get("type").asText();
 		final AssetType assetType = AssetType.getAssetType(assetTypeStr, mapper);
 
-		TerariumAsset asset = mapper.treeToValue(node.get("asset"), assetType.getAssetClass());
+		final TerariumAsset asset = mapper.treeToValue(node.get("asset"), assetType.getAssetClass());
 
 		Map<String, FileExport> files = new HashMap<>();
-		if (node.has("files")) {
+		if (node.has("files") && !node.get("files").isNull()) {
```
Comment (YohannParis) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/AssetExportDeserializer.java:27]: What an incongrus test. How can node.files exist but return null?

**Review by YohannParis — APPROVED** [review]

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/AssetExportDeserializer.java (hunk 1 above)
Comment (bigglesandginger) [reply packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/AssetExportDeserializer.java:27]: Should that be an OR and not an AND (like ContentTypeDeserializer L23)?

Comment (bigglesandginger) [reply packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/AssetExportDeserializer.java:27]: Wait the logic in the two is totally confusing :). NOTs abound... .

Comment (YohannParis) [reply packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/AssetExportDeserializer.java:27]: _ChatGPT_ says to use `if (node.hasNonNull("files")) {` instead.

**Review by bigglesandginger — APPROVED** [review]

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/AssetExportDeserializer.java (hunk 1 above)
Comment (kbirk) [reply packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/AssetExportDeserializer.java:27]: I don't understand why it worked before, but it is returning a `NullNode` entry now, makes the `has` return true, but you get an exception if you try to use the entry

Comment (kbirk) [reply packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/AssetExportDeserializer.java:27]: I like the `hasNonNull` ill use that instead

**Comment by bigglesandginger** [comment]
Javascript, making its namesake parent garbage through association too.  Amazing, well done whoever coded the JSON libraries.

**Pull request merged**